adminSiteClient/VariablesAnnotationTypesAndUtils.tsx

```diff
@@ -183,3 +273,244 @@ export const getItemStyle = (
     // styles we need to apply on draggables
     ...draggableStyle,
 })
+
+export function isConfigColumn(columnName: string): boolean {
+    return columnName.startsWith("/")
+}
+
+export const filterPanelInitialConfig: BasicConfig = AntdConfig as BasicConfig
+export const initialFilterQueryValue: JsonGroup = {
+    id: QbUtils.uuid(),
+    type: "group",
+}
+export type FilterPanelState = {
+    tree: ImmutableTree
+    config: Config
+}
+
+export function getLogicOperator(str: string): BinaryLogicOperators {
+    if (str === "AND") return BinaryLogicOperators.and
+    else if (str === "OR") return BinaryLogicOperators.or
+    else throw Error(`unknown logic operator: ${str}`)
+}
+
+export function getComparisonOperator(
+    str: string
+): ComparisonOperator | undefined {
+    return match(str)
+        .with("less", () => ComparisonOperator.less)
+        .with("less_or_equal", () => ComparisonOperator.lessOrEqual)
+        .with("greater", () => ComparisonOperator.greater)
+        .with("greater_or_equal", () => ComparisonOperator.greaterOrEqual)
+        .otherwise(() => undefined)
+}
+
+export function getNullCheckOperator(
+    str: string
+): NullCheckOperator | undefined {
+    return match(str)
+        .with("is_null", () => NullCheckOperator.isNull)
+        .with("is_not_null", () => NullCheckOperator.isNotNull)
+        .otherwise(() => undefined)
+}
+
+export function getFieldSymbol(fieldName: string): Operation {
+    if (isConfigColumn(fieldName)) return new JsonPointerSymbol(fieldName)
+    else
+        return new SqlColumnName(
+            readOnlyColumnNamesFields.get(fieldName)!.sExpressionColumnTarget
+        )
+}
+
+export function getValueAtom(val: any): Operation | undefined {
+    if (typeof val === "string") return new StringAtom(val)
+    else if (typeof val === "number") return new NumberAtom(val)
+    else if (typeof val === "boolean") return new BooleanAtom(val)
+    else return undefined
+}
+
+export function getEqualityOperator(str: string): EqualityOperator | undefined {
+    if (str === "equal" || str === "select_equals")
+        return EqualityOperator.equal
+    else if (str === "not_equal" || str === "select_not_equals")
+        return EqualityOperator.unequal
+    else return undefined
+}
+
+export function filterTreeToSExpression(
+    filterTree: JsonItem
+): Operation | undefined {
+    if (filterTree.type === "group") {
+        // If we have a group then we need to decide
+        // on the operator and build the list of children recursively
+        const logicOperator = getLogicOperator(
+            filterTree.properties?.conjunction ?? "AND"
+        )
+
+        let children: Operation[] = []
+        if (isArray(filterTree.children1))
```

Comment:
I think you are right but the type definition claims otherwise. This is probably an error in the type definition though. I'm tempted to leave it as is but add a comment, what do you think?

Reply:
Either is fine with me - either leaving it in as-is or going with a `console.error("This code path is not supposed to happen")` instead.